sympy/polys/numberfields/minpoly.py

```diff
@@ -571,11 +571,17 @@ def _minpoly_compose(ex, x, dom):
         return x - ex
 
     if dom.is_QQ and _is_sum_surds(ex):
+        # extract gcd's from all radicals to confirm
+        # non-zero value
+        ex = _mexpand(factor_terms(ex, radical=True))
+        if not ex:
+            return x
         # eliminate the square roots
         ex -= x
         while 1:
             ex1 = _separate_sq(ex)
             if ex1 is ex:
+                assert ex.is_Add, 'unexpected return value'
                 return ex
```

Comment:
The obvious fix here is to call `factor` as is done for all other cases in `minpoly`.

If we are to avoid calling factor then I think we need a proof that it is not necessary somehow.

Reply:
factoring is now used and a pertinent test is added